File: Kotlin/성격_유형_검사하기.kt
fun solution(survey: Array<String>, choices: IntArray): String {
    val map = mutableMapOf<Char,Int>()
    var result = ""
    val arr = arrayOf('R', 'T', 'C', 'F', 'J', 'M', 'A', 'N')
    arr.forEach { map[it] = 0 }
    survey.forEachIndexed {index, it -> if(choices[index] - 4 <= 0) map[it.first()] = map[it.first()]!! + 4 - choices[index]  else map[it.last()] = map[it.last()]!! + choices[index] - 4}
    result += if(map.getValue('R') >= map.getValue('T')) "R" else "T"
    result += if(map.getValue('C') >= map.getValue('F')) "C" else "F"
    result += if(map.getValue('J') >= map.getValue('M')) "J" else "M"
    result += if(map.getValue('A') >= map.getValue('N')) "A" else "N"
    println(map)
    return result
}

File: JavaScript/성격_유형_검사하기.js
function solution(survey, choices) {
  let scoreBoard = [3, 2, 1, 0, 1, 2, 3];
  let data = {
    R: 0,
    T: 0,
    C: 0,
    F: 0,
    J: 0,
    M: 0,
    A: 0,
    N: 0,
  };
  survey.forEach(([a, b], i) => {
    choices[i] > 3
      ? (data[b] += scoreBoard[choices[i] - 1])
      : (data[a] += scoreBoard[choices[i] - 1]);
  });
  let answer = "";
  answer += data["R"] >= data["T"] ? "R" : "T";
  answer += data["C"] >= data["F"] ? "C" : "F";
  answer += data["J"] >= data["M"] ? "J" : "M";
  answer += data["A"] >= data["N"] ? "A" : "N";
  return answer;
}
console.log(solution(["TR", "RT", "TR"], [7, 1, 3]));
